repytah/utilities.py

```diff
@@ -212,83 +206,63 @@ def find_initial_repeats(thresh_mat, bandwidth_vec, thresh_bw):
                     # NOTE: the middle interval may NOT exist
                     # Vector of 1's that is the length of the number of
                     # overlapping intervals. This is used a lot.
-                    ones_no = np.ones(num_overlaps)
+                    ones_no = np.ones(num_overlaps, int)
 
                     # 2a) Left Overlap
                     # Width is the same for corresponding left and right overlaps
-                    K = start_j_shin - start_i_shin
+                    K = start_j_over - start_i_over
 
-                    # List pairs of starts with their ends and widths
-                    i_sshin = np.vstack((start_i_shin, (start_j_shin -
-                                                           ones_no))).T
-                    j_sshin = np.vstack((start_j_shin, (start_j_shin +
-                                                           K - ones_no))).T
-                    sint_lst = np.column_stack((i_sshin, j_sshin, K.T))
+                    smatch_i_over = start_j_over - 1
+                    smatch_j_over = start_j_over + K - 1
 
-                    i_s = np.argsort(K)  # Return the indices that would sort K
-                    sint_lst = sint_lst[i_s, ]
+                    # List pairs of starts with their ends and widths
+                    sint_lst = np.vstack((start_i_over, smatch_i_over, 
+                        start_j_over, smatch_j_over, K)).T
 
                     # Remove the pairs that fall below the bandwidth threshold
-                    cut_s = np.argwhere((sint_lst[:, 4] > thresh_bw))
-                    cut_s = cut_s.T
-                    sint_lst = sint_lst[cut_s][0]
+                    keep_s = sint_lst[:, 4] > thresh_bw
+                    sint_lst = sint_lst[keep_s]
 
                     # Add the new left overlapping intervals to the full list
                     # of left overlapping intervals
                     sint_all = np.vstack((sint_all, sint_lst))
 
                     # 2b) Right Overlap
 
-                    end_i_shin = start_i_shin + (bw - 1)
-                    end_j_shin = start_j_shin + (bw - 1)
+                    end_i_over = start_i_over + (bw - 1)
+                    end_j_over= start_j_over + (bw - 1)
 
-                    i_eshin = np.vstack((end_i_shin + ones_no - K,
-                                         end_i_shin)).T
-                    j_eshin = np.vstack((end_i_shin + ones_no,
-                                         end_j_shin)).T
-                    eint_lst = np.column_stack((i_eshin, j_eshin, K.T))
+                    ematch_i_over = end_i_over - K + 1 
+                    ematch_j_over = end_i_over - 1
```

Comment:
why is it `ematch_j_over = end_i_over - 1` and not `ematch_j_over = end_i_over + 1`

Reply:
This was an error on my part